### PR TITLE
Login/Registration

### PR DESCRIPTION
**Pull request by avminon**
-includes auth and seeders
-routes, views, controllers
-no assets included

### CODE REVIEW

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+     * Display the posts of a particular user
+	 *
+	 * @param int $id
+	 * @return Response
+	*/
+    public function index()
+    {
+        $title = 'Title';
+        $user = User::find(\Auth::id());
+        //return view('home')->withTitle($title)->with('user',$user)->with('activities', \Auth::user()->id);
```
Comment (nguyenthanhtung88) [inline app/Http/Controllers/UserController.php]: Remove unnecessary line +_+

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    
+    /*
+     * Display the posts of a particular user
+	 *
+	 * @param int $id
+	 * @return Response
+	*/
+    public function index()
+    {
+        $title = 'Title';
```
Comment (nguyenthanhtung88) [inline app/Http/Controllers/UserController.php]: You should try to use Localization feature (https://laravel.com/docs/5.1/localization)
Don't use static text here :bow:

File: app/Http/Controllers/UserController.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+class UserController extends Controller
+{
+    /**
+     * Create a new controller instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        $this->middleware('auth');
```
Comment (nguyenthanhtung88) [inline app/Http/Controllers/UserController.php]: You should define `Middleware` in route file for this case.
Take a look to https://laravel.com/docs/5.1/routing#route-groups

File: app/Http/Controllers/UserController.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    /*
+     * Display the posts of a particular user
+	 *
+	 * @param int $id
+	 * @return Response
+	*/
+    public function index()
+    {
+        $title = 'Title';
+        $user = User::find(\Auth::id());
```
Comment (nguyenthanhtung88) [inline app/Http/Controllers/UserController.php]: Use `\Auth::user()` for user object

File: app/Http/Controllers/UserController.php [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        //return view('home')->withTitle($title)->with('user',$user)->with('activities', \Auth::user()->id);
+        return view('home')->withTitle($title)->with('user',$user);
```
Comment (nguyenthanhtung88) [inline app/Http/Controllers/UserController.php]: Use `return view('home', compact('title', 'user'));` is better.

File: app/Http/Controllers/UserController.php [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        return view('admin.profile', $data);
+    }
+
+    public function update(Request $request)
+    {
+        $this->validate($request, [
+            'name' => 'required|max:255',
+        ]);
+        $user_id = $request->input('user_id');
+        $new_name = $request->input('name');
```
Comment (nguyenthanhtung88) [inline app/Http/Controllers/UserController.php]: Using `camelCase` for variable name instead of `snake_case`

File: app/Http/Controllers/UserController.php [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+        $user = Users::find($user_id);
+        $user->name = $new_name;
+        $user_>save();
```
Comment (nguyenthanhtung88) [inline app/Http/Controllers/UserController.php]: why `_` ???

File: app/Http/Controllers/UserController.php [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+        $data['user'] = User::find($id);
+        return view('admin.profile', $data);
+    }
+
+    public function update(Request $request)
+    {
+        $this->validate($request, [
+            'name' => 'required|max:255',
+        ]);
+        $user_id = $request->input('user_id');
```
Comment (nguyenthanhtung88) [inline app/Http/Controllers/UserController.php]: `camelCase`

File: app/Http/routes.php [hunk 9]
```diff
@@ -10,7 +10,14 @@
 | and give it the controller to call when that URI is requested.
 |
 */
+Route::get('/',['as' => 'home', 'uses' => 'UserController@index']);
+
+Route::get('/home',['as' => 'home', 'uses' => 'UserController@index']);
```
Comment (nguyenthanhtung88) [inline app/Http/routes.php]: one space after `,`

File: app/Http/routes.php [hunk 10]
```diff
@@ -10,7 +10,14 @@
 | and give it the controller to call when that URI is requested.
 |
 */
+Route::get('/',['as' => 'home', 'uses' => 'UserController@index']);
```
Comment (nguyenthanhtung88) [inline app/Http/routes.php]: one space after `,`

File: app/Http/routes.php [hunk 11]
```diff
@@ -10,7 +10,14 @@
...
+
+Route::get('/home',['as' => 'home', 'uses' => 'UserController@index']);
+
+Route::controllers([
+    'auth' => 'Auth\AuthController',
+    'password' => 'Auth\PasswordController',
+]);
+
+Route::get('user/{id}', 'UserController@profile')->where('id', '[0-9]+');
```
Comment (nguyenthanhtung88) [inline app/Http/routes.php]: `^[0-9]+$` for ensure that user id is numeric

File: database/seeds/DatabaseSeeder.php [hunk 12]
```diff
@@ -9,13 +9,11 @@ class DatabaseSeeder extends Seeder
...
      * @return void
-     */
+    */
     public function run()
     {
         Model::unguard();
-
-        // $this->call(UserTableSeeder::class);
-
+        $this->call('UsersTableSeeder');
```
Comment (nguyenthanhtung88) [inline database/seeds/DatabaseSeeder.php]: Should use `UserTableSeeder::class`

File: database/seeds/UsersTableSeeder.php [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+     *
+     * @return void
+     */
+    public function run()
+    {
+        DB::table('users')->insert([
+            'name'=>'admin',
+            'email'=>'admin@email.com',
+            'password'=>bcrypt('admin'),
+            'type'=>'admin',
```
Comment (nguyenthanhtung88) [inline database/seeds/UsersTableSeeder.php]: Should have 1 space before and after `=>`. Please check others above.

File: resources/views/app.blade.php [hunk 14]
```diff
@@ -0,0 +1,119 @@
...
+                        </li>
+                        @if ((!Auth::guest()) && ($user->isAdmin()))
+                            <li>
+                                <a href="{{ url('/') }}">Words</a>
+                            </li>
+                            <li>
+                                <a href="{{ url('categories/') }}">Categories</a>
+                            </li>
+                          @endif
+
```
Comment (nguyenthanhtung88) [inline resources/views/app.blade.php]: Remove this line

File: resources/views/app.blade.php [hunk 15]
```diff
@@ -0,0 +1,119 @@
...
+                            <a href="{{ url('/') }}">Home</a>
+                        </li>
+                        @if ((!Auth::guest()) && ($user->isAdmin()))
+                            <li>
+                                <a href="{{ url('/') }}">Words</a>
+                            </li>
+                            <li>
+                                <a href="{{ url('categories/') }}">Categories</a>
+                            </li>
+                          @endif
```
Comment (nguyenthanhtung88) [inline resources/views/app.blade.php]: Please using same indent with `@if` statement above

File: resources/views/app.blade.php [hunk 16]
```diff
@@ -0,0 +1,119 @@
...
+                        </li>
+                        <li>
+                            <a href="{{ url('/auth/register') }}">Register</a>
+                        </li>
+                        @else
+                        <li class="dropdown">
+                            <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">{{ Auth::user()->name }} <span class="caret"></span></a>
+                            <ul class="dropdown-menu" role="menu">
+                                <li>
+                                    <a href="{{ url('/user/'.Auth::id()) }}">My Profile</a>
```
Comment (nguyenthanhtung88) [inline resources/views/app.blade.php]: `url('/user/'.Auth::id())`
Insert 1 space before and after `.`

File: resources/views/auth/login.blade.php [hunk 17]
```diff
@@ -0,0 +1,61 @@
...
+                                        <input type="checkbox" name="remember"> Remember Me
+                                    </label>
+                                </div>
+                            </div>
+                        </div>
+
+                        <div class="form-group">
+                            <div class="col-md-6 col-md-offset-4">
+                                <button type="submit" class="btn btn-primary">Login</button>
+
```
Comment (nguyenthanhtung88) [inline resources/views/auth/login.blade.php]: Remove this line

File: resources/views/home.blade.php [hunk 18]
```diff
@@ -0,0 +1,39 @@
+@extends('app')
+
+@section('title')
+    @if (Auth::guest())
+        Welcome!
+    @endif
+    @if (Auth::check())
+        Welcome {{$title}}!
```
Comment (nguyenthanhtung88) [inline resources/views/home.blade.php]: `{{$title}}` => need space

File: resources/views/home.blade.php [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+    @endif
+@endsection
+
+
+@section('content')
+    @if (Auth::check())
+        @if ($user->isAdmin())
+            Admin
+        @endif
+                <div class="panel-heading">Dashboard</div>
```
Comment (nguyenthanhtung88) [inline resources/views/home.blade.php]: Please use same indent with `@endif` statement above.
Check another indent mistake below.

File: resources/views/app.blade.php [hunk 20]
```diff
@@ -0,0 +1,119 @@
...
+                    </button>
+                    <a class="navbar-brand" href="http://www.framgia.com/jp">Framgia</a>
+                </div>
+
+                <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
+                    <ul class="nav navbar-nav">
+                        <li>
+                            <a href="{{ url('/') }}">Home</a>
+                        </li>
+                        @if ((!Auth::guest()) && ($user->isAdmin()))
```
Comment (nguyenthanhtung88) [inline resources/views/app.blade.php]: Please try to use Laravel Helper, in this case: https://laravel.com/docs/5.1/helpers#method-auth
`auth()->guest()` is better

File: resources/views/app.blade.php [hunk 21]
```diff
@@ -0,0 +1,119 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="utf-8">
+        <meta http-equiv="X-UA-Compatible" content="IE=edge">
+        <meta name="viewport" content="width=device-width, initial-scale=1">
+        <title>Framgia E-Learning System</title>
```
Comment (nguyenthanhtung88) [inline resources/views/app.blade.php]: Use Localization for static text, you can get acquainted with awesome technical :+1: 
The advantage is:
- Re-use your text if we have in multiple view
- Translation for multiple language
==> It's good habbit, i think you should give a try :smiley:

File: resources/views/app.blade.php [hunk 22]
```diff
@@ -0,0 +1,119 @@
...
+                            </ul>
+                        </li>
+                        @endif
+                    </ul>
+                </div>
+            </div>
+        </nav>
+
+        <div class="container">
+            @if (Session::has('message'))
```
Comment (nguyenthanhtung88) [inline resources/views/app.blade.php]: Please use `Session Helper`: https://laravel.com/docs/5.1/helpers#method-session

File: resources/views/app.blade.php [hunk 23]
```diff
@@ -0,0 +1,119 @@
...
+            <div class='flash alert-danger'>
+                <ul class="panel-body">
+                    @foreach ( $errors->all() as $error )
+                    <li>
+                        {{ $error }}
+                    </li>
+                    @endforeach
+                </ul>
+            </div>
+            @endif
```
Comment (nguyenthanhtung88) [inline resources/views/app.blade.php]:
```
@if (Session::has('message'))
            <div class="flash alert-info">
                <p class="panel-body">
                    {{ Session::get('message') }}
                </p>
            </div>
            @endif
            @if ($errors->any())
            <div class='flash alert-danger'>
                <ul class="panel-body">
                    @foreach ( $errors->all() as $error )
                    <li>
                        {{ $error }}
                    </li>
                    @endforeach
                </ul>
            </div>
            @endif
```
I saw this code you use in multiple view, please follow DRY (Dont-Repeat-Yourself)
=> You can with these code using `partial` technical, take a look at https://laraveltips.wordpress.com/2015/06/17/view-partials-in-laravel-5-1/

File: resources/views/auth/login.blade.php [hunk 24]
```diff
@@ -0,0 +1,61 @@
...
+                    @if (count($errors) > 0)
+                        <div class="alert alert-danger">
+                            <strong>Whoops!</strong> There were some problems with your input.<br><br>
+                            <ul>
+                                @foreach ($errors->all() as $error)
+                                    <li>{{ $error }}</li>
+                                @endforeach
+                            </ul>
+                        </div>
+                    @endif
```
Comment (nguyenthanhtung88) [inline resources/views/auth/login.blade.php]: Please use `partial view`

File: app/Http/Controllers/UserController.php [hunk 25]
```diff
@@ -0,0 +1,58 @@
...
+        return view('home')->withTitle($title)->with('user',$user);
+    }
+
+    public function profile(Request $request, $id)
+    {
+        $data['user'] = User::find($id);
+        return view('admin.profile', $data);
+    }
+
+    public function update(Request $request)
```
Comment (nguyenthanhtung88) [inline app/Http/Controllers/UserController.php]: I can not see any route define for this function? why you put on this pull request ???

File: app/Http/Controllers/UserController.php [hunk 26]
```diff
@@ -0,0 +1,58 @@
...
+	*/
+    public function index()
+    {
+        $title = 'Title';
+        $user = User::find(\Auth::id());
+        //return view('home')->withTitle($title)->with('user',$user)->with('activities', \Auth::user()->id);
+        return view('home')->withTitle($title)->with('user',$user);
+    }
+
+    public function profile(Request $request, $id)
```
Comment (nguyenthanhtung88) [inline app/Http/Controllers/UserController.php]: What does this function focus on? Show profile of particular user right?
Please try to use `Resource Controller` (https://laravel.com/docs/5.1/controllers#restful-resource-controllers)
If you want to show user's profile, you can use `ProfileController` or advanced technical you can define `UserProfileController` for nested resource :+1: 

Please remember that we should follow `resource controller` and 1 resource controller should contain only 7 actions

File: app/Http/Controllers/UserController.php [hunk 27]
```diff
@@ -0,0 +1,58 @@
...
+    {
+        $data['user'] = User::find($id);
+        return view('admin.profile', $data);
```
Comment (nguyenthanhtung88) [inline app/Http/Controllers/UserController.php]: I cannot see any view include in your PR +_+

File: app/Http/Controllers/UserController.php (hunk 26 above)
Comment (naid) [inline app/Http/Controllers/UserController.php]: Sorry, this will be removed as the views associated with it is not yet included

File: app/Http/Controllers/UserController.php [hunk 28]
```diff
@@ -0,0 +1,35 @@
...
+use Illuminate\Http\Request;
+
+class UserController extends Controller
+{
+    /**
+     * Create a new controller instance.
+     *
+     * @return void
+     */
+    public function __construct()
```
Comment (nguyenthanhtung88) [inline app/Http/Controllers/UserController.php]: You should remove blank function +_+ it's not necessary

File: resources/lang/en/common.php [hunk 29]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+return [
+    'main' => [
+        'page_title' =>  'Framgia E-Learning System',
+
+    ],
+
+    'users' => [
+        'page_title' =>	'Title',
```
Comment (nguyenthanhtung88) [inline resources/lang/en/common.php]: We have too much space here after `=>` +_+

File: resources/lang/en/common.php [hunk 30]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+return [
+    'main' => [
+        'page_title' =>  'Framgia E-Learning System',
+
+    ],
+
```
Comment (nguyenthanhtung88) [inline resources/lang/en/common.php]: I think we don't need this line

File: resources/views/emails/password.blade.php [hunk 31]
```diff
@@ -0,0 +1 @@
+Click here to reset your password: {{ url('password/reset/'.$token) }}
```
Comment (nguyenthanhtung88) [inline resources/views/emails/password.blade.php]: `'password/reset/'.$token` => need space after and before `.`

File: resources/views/home.blade.php [hunk 32]
```diff
@@ -0,0 +1,38 @@
...
+@section('content')
+    @if (Auth::check())
+        @if ($user->isAdmin())
+            Admin
+        @endif
+            <div class="panel-heading">Dashboard</div>
+
+            <div class="panel-body">
+                <div class="col-md-3">
+                    <table height="250" border="1px" width="100%">
```
Comment (nguyenthanhtung88) [inline resources/views/home.blade.php]: Why we have blank table here ???

File: resources/views/home.blade.php [hunk 33]
```diff
@@ -0,0 +1,38 @@
+@extends('app')
+
+@section('title')
+    @if (Auth::guest())
```
Comment (nguyenthanhtung88) [inline resources/views/home.blade.php]: Why you don't use `auth()` helper like you used above :cry:

File: resources/views/app.blade.php [hunk 34]
```diff
@@ -0,0 +1,118 @@
...
+                    <a class="navbar-brand" href="http://www.framgia.com/jp">Framgia</a>
+                </div>
+
+                <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
+                    <ul class="nav navbar-nav">
+                        <li>
+                            <a href="{{ url('/') }}">Home</a>
+                        </li>
+                        @if ((!auth()->guest()) && ($user->isAdmin()))
+                              <li>
```
Comment (nguyenthanhtung88) [inline resources/views/app.blade.php]: Check indent +_+

File: resources/views/home.blade.php (hunk 32 above)
Comment (naid) [inline resources/views/home.blade.php]: it's a placeholder for the avatar but we can remove it for now since it's not really part of the feature. sorry about that.

File: app/Http/Controllers/UserController.php [hunk 35]
```diff
@@ -0,0 +1,26 @@
...
+use Illuminate\Http\Request;
+
+class UserController extends Controller
+{
+    /*
+     * Display the posts of a particular user
+	 *
+	 * @param int $id
+	 * @return Response
+	*/
```
Comment (nguyenthanhtung88) [inline app/Http/Controllers/UserController.php]: This description is not related to this function, and the indent is not good.
Please remove this if you don't need

File: resources/lang/en/common.php [hunk 36]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+return [
+    'main' => [
+        'page_title' => 'Framgia E-Learning System',
+
+    ],
+    'users' => [
+        'page_title' =>	'Title',
```
Comment (nguyenthanhtung88) [inline resources/lang/en/common.php]: Too many space after `=>`

File: resources/views/shared/flash_message.blade.php [hunk 37]
```diff
@@ -0,0 +1,18 @@
...
+    <div class="flash alert-info">
+        <p class="panel-body">
+            {{ Session::get('message') }}
+        </p>
+    </div>
+@endif
+@if ($errors->any())
+    <div class='flash alert-danger'>
+        <ul class="panel-body">
+            @foreach ( $errors->all() as $error )
```
Comment (nguyenthanhtung88) [inline resources/views/shared/flash_message.blade.php]: Please follow coding convention for `foreach` loop (https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md#55-foreach)

File: resources/views/shared/flash_message.blade.php [hunk 38]
```diff
@@ -0,0 +1,18 @@
...
+    <div class="flash alert-info">
+        <p class="panel-body">
+            {{ Session::get('message') }}
+        </p>
+    </div>
+@endif
+@if ($errors->any())
+    <div class='flash alert-danger'>
+        <ul class="panel-body">
+            @foreach ( $errors->all() as $key => $error )
```
Comment (nguyenthanhtung88) [inline resources/views/shared/flash_message.blade.php]: Sorry but it's not what i mention in previous comment :cry: 
1. You don't use `$key` so we don't need here.
2. The format should be `@foreach ($errors->all() as $error)`

**Comment by nguyenthanhtung88** [comment]
OK
